libsplinter/src/network/auth/handlers/v1_handlers.rs

```diff
@@ -141,9 +157,32 @@ impl Handler for AuthProtocolRequestHandler {
                     version
                 );
 
+                let mut accepted_authorization_type = vec![];
+                #[cfg(feature = "trust-authorization")]
+                {
+                    accepted_authorization_type.push(PeerAuthorizationType::Trust);
+                }
+
+                #[cfg(feature = "challenge-authorization")]
+                match self.expected_authorization {
```

Comment:
If we have an expected_authorization type, that means we are the side that has circuit/proposal and we need to make sure that we only send the authorization type that is required, otherwise the other side (which does not yet have a circuit/proposal information) could choose the wrong type of authorization. If we do not have an expected authorization type we want to include all of the supported authorization types so the other side can make the decision on what type of authorization to do. I was having trouble makes this work cleanly with the two different traits that needed to be supported.

Reply:
^ could be made into a code comment

Reply:
Adding the code comment

Reply:
added